src/App.js

```diff
@@ -1,16 +1,43 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+  const [likeCount, setLikeCount] = useState(0);
+  
+  const updateLike = (id) => {
+    setMessages(messages => {
+      return messages.map(message => {
+        if (id === message.id) {
+          return {
+            ...message,
+            liked: !message.liked,
+          }
+        } else {
+          return message;
+        }
+      });
+    });
+  };
+
+  const countLikes = ((liked) => {
+    if (liked) {
+      setLikeCount(() => likeCount + 1);
```

Comment:
In general, when you're updating state that depends directly on the previous state, you should use the argument that React passes to the state update function's callback argument. For example,
```js
setLikeCount((previousLikeCount) => previousLikeCount + 1);
```
That way you'll be guaranteed to have the computation based on the actual most recent state.